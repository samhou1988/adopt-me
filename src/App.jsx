import React, { useState } from 'react';
import { BrowserRouter, Routes, Route, Link } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import AdoptedPetContext from './AdoptedPetContext';
import Details from './Details';
import SearchParams from './SearchParams';

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      staleTime: Infinity,
      cacheTime: Infinity,
    },
  },
});

export default function App() {
  const adoptedPet = useState(null);
  return (
    <React.StrictMode>
      <BrowserRouter>
        <AdoptedPetContext.Provider value={adoptedPet}>
          <QueryClientProvider client={queryClient}>
            <header>
              <Link to="/">Adopt Me!</Link>
            </header>
            <Routes>
              <Route path="/details/:id" element={<Details />} />
              <Route path="/" element={<SearchParams />} />
            </Routes>
          </QueryClientProvider>
        </AdoptedPetContext.Provider>
      </BrowserRouter>
    </React.StrictMode>
  );
}
