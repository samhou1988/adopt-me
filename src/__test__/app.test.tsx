import React from 'react'
import renderer from 'react-test-renderer';
import { expect, test } from 'vitest'

import App from '../App'

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  return result as renderer.ReactTestRendererJSON
}

test('App', () => {
  const component = renderer.create(
    <App />,
  )
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
