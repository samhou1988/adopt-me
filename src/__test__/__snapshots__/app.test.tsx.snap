// Vitest Snapshot v1

exports[`App 1`] = `
[
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      Adopt Me!
    </a>
  </header>,
  <div
    className="search-params"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="location"
      >
        Location
        <input
          id="location"
          name="location"
          placeholder="Location"
        />
      </label>
      <label
        htmlFor="animal"
      >
        Animal
        <select
          id="animal"
          name="animal"
          onBlur={[Function]}
          onChange={[Function]}
        >
          <option />
          <option
            value="bird"
          >
            bird
          </option>
          <option
            value="cat"
          >
            cat
          </option>
          <option
            value="dog"
          >
            dog
          </option>
          <option
            value="rabbit"
          >
            rabbit
          </option>
          <option
            value="reptile"
          >
            reptile
          </option>
        </select>
      </label>
      <label
        htmlFor="breed"
      >
        Breed
        <select
          id="breed"
          name="breed"
        >
          <option>
            Please choose...
          </option>
        </select>
      </label>
      <button>
        Submit
      </button>
    </form>
    <div
      className="search"
    >
      <h1>
        No Pets Found
      </h1>
    </div>
  </div>,
]
`;

exports[`Link changes the class when hovered 1`] = `
[
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      Adopt Me!
    </a>
  </header>,
  <div
    className="search-params"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="location"
      >
        Location
        <input
          id="location"
          name="location"
          placeholder="Location"
        />
      </label>
      <label
        htmlFor="animal"
      >
        Animal
        <select
          id="animal"
          name="animal"
          onBlur={[Function]}
          onChange={[Function]}
        >
          <option />
          <option
            value="bird"
          >
            bird
          </option>
          <option
            value="cat"
          >
            cat
          </option>
          <option
            value="dog"
          >
            dog
          </option>
          <option
            value="rabbit"
          >
            rabbit
          </option>
          <option
            value="reptile"
          >
            reptile
          </option>
        </select>
      </label>
      <label
        htmlFor="breed"
      >
        Breed
        <select
          id="breed"
          name="breed"
        >
          <option>
            Please choose...
          </option>
        </select>
      </label>
      <button>
        Submit
      </button>
    </form>
    <div
      className="search"
    >
      <h1>
        No Pets Found
      </h1>
    </div>
  </div>,
]
`;
